controllers/custom/custom_controller.go

```diff
@@ -108,7 +109,7 @@ type Request struct {
 }
 
 // Starts the low level controller
-func (c *CustomController) Start(stop <-chan struct{}) error {
+func (c *CustomController) Start(ctx context.Context) error {
```

Comment:
I think this is broken from the start, context.TODO() and context.Background() both return an empty context. We need to start using `ctrl.SetupSignalHandler()` for all the contexts.